crates/grafbase-sdk/src/host_io/postgres/types.rs

```diff
@@ -0,0 +1,378 @@
+//! # Database Type conversions for Postgres
+//!
+//! This module provides traits and implementations for converting Rust types
+//! to Postgres database values.
+//!
+//! ## Key Components
+//!
+//! - `DatabaseType` trait: Defines how types can be converted to database values
+//! - `DatabaseValue`: Represents a value bound to a SQL query parameter
+//! - Type implementations: Conversions for Rust primitives and common types
+use std::time::SystemTime;
+
+use chrono::Utc;
+use uuid::Uuid;
+
+use crate::wit;
+
+/// A trait for types that can be converted to database values.
+///
+/// This trait allows Rust types to be converted into database values that can be
+/// used in SQL queries. It provides methods to convert values with or without
+/// explicit type hints.
+pub trait DatabaseType {
```

Comment:
Do you want to expose that type publicly or should it be kept private, as in sealed/name not accessible?
You're using raw wit types, so doesn't seem like something users should use, but 🤷

Reply:
yeah just wanted to point it out just in case^^

Reply:
I know. Kind of think this could be public, but then the API needs to change a bit. Will figure it out later.